src/libraries/System.IO.Compression.Brotli/src/System/IO/Compression/BrotliUtils.cs

```diff
@@ -11,7 +11,7 @@ internal static partial class BrotliUtils
         public const int Quality_Min = 0;
         public const int Quality_Default = 4;
         public const int Quality_Max = 11;
-        public const int MaxInputSize = int.MaxValue - 515; // 515 is the max compressed extra bytes
+        public const int MaxInputSize = int.MaxValue - 524_166; // 524_166 is the max compressed extra bytes
```

Comment:
Where does this number come from?

Reply:
Before `int.MaxValue - 515 = 2_147_483_132` was producing max compressed length, i.e. `GetMaxCompressedLength(2_147_483_132))` returned int.MaxValue. Now this produces negative number 

After testing with various values now this value equals `524_166` and GetMaxCompressedLength(int.MaxValue - 524_166))` produces `int.MaxValue`

Reply:
Rather than having this hardcoded constant, though, couldn't we just call the native method and then throw ArgumentOutOfRangeException if the result was > int.MaxValue? Part of the point of this change was delegating all behavior to the native function, but we're lifting some of the behavior back up into the C# by hardcoding this constant, which in theory could be invalidated the next time we take a Brotli update.

Reply:
Right, good point